src/lib/fetch.ts

```diff
@@ -188,6 +219,17 @@ async function fetchMissingData(graphqlEndpoint: string) {
       if (response.error === undefined) delete accountsToFetch[key];
     }
   );
+  let actionPromises = Object.entries(actionsToFetch).map(
+    async ([key, { publicKey, tokenId }]) => {
+      let response = await fetchActions(
+        { publicKey, tokenId },
+        archiveEndpoint
+      );
+      if ('error' in response && response.error === undefined)
+        delete actionsToFetch[key];
```

Comment:
this logic seems wrong to me: we want to delete from `actionsToFetch` if and only if the fetch was successful. that means checking for `if (response.error === undefined)`, or, if TS doesn't like that, `if (!('error' in response) || response.error === undefined)`